### ISSUE TITLE
[Failing test][sig-cloud-provider] gce-cos-master-serial

### ISSUE THREAD

**Issue by RinkiyaKeDad** [kubernetes/kubernetes#106008]
### Which jobs are failing?

gce-cos-master-serial

### Which tests are failing?

- Kubernetes e2e suite.[sig-storage] [Serial] Volume metrics Ephemeral should create prometheus metrics for volume provisioning and attach/detachChanges
- Kubernetes e2e suite.[sig-storage] [Serial] Volume metrics Ephemeral should create prometheus metrics for volume provisioning errors [Slow]Changes
- Kubernetes e2e suite.[sig-storage] [Serial] Volume metrics PVC should create prometheus metrics for volume provisioning and attach/detachChanges
- Kubernetes e2e suite.[sig-storage] [Serial] Volume metrics PVC should create prometheus metrics for volume provisioning errors [Slow]Changes
- e2e.go: Test 

### Since when has it been failing?

2021-10-27 03:36 PDT

### Testgrid link

https://testgrid.k8s.io/sig-release-master-informing#gce-cos-master-serial

### Reason for failure (if possible)

For the first four failing tests:

```
Error fetching c-m updated storage metrics
Expected
    <int>: 0
not to equal
    <int>: 0
```

For e2e.go:

```
error during ./hack/ginkgo-e2e.sh --ginkgo.focus=\[Serial\]|\[Disruptive\] --ginkgo.skip=\[Driver:.gcepd\]|\[Flaky\]|\[Feature:.+\] --minStartupPods=8 --report-dir=/workspace/_artifacts --disable-log-dump=true: exit status 1
```

### Relevant SIG(s)

/sig cloud-provider

**Comment by aojea** [comment]
seems a dup of https://github.com/kubernetes/kubernetes/pull/105970
cc @pohly

**Comment by mauriciopoppe** [comment]
There are a few more recent failures in that job related with turning on the CSIMigrationGCE feature flag in https://github.com/kubernetes/kubernetes/pull/104722,@leiyiz is fixing them in https://github.com/kubernetes/kubernetes/pull/106503

**Comment by mauriciopoppe** [comment]
gce-cos-master-serial should be all green after https://github.com/kubernetes/kubernetes/pull/106503 is merged and it's because it's skipping the test that are failing, please don't close this issue yet, I'll post a comment once the fix for these failures is ready, the hope in the future is that the skipped tests will run with `kubernetes/cloud-provider-gcp` bringing up the e2e cluster instead of it coming from `kubernetes/kubernetes` which is no longer installing the required resources for these failing tests anymore.

**Comment by mauriciopoppe** [comment]
/close

Fixed in #106609, testgrid for kubernetes shows that the tests are skipped but they passed when I set it up with an external CSI Driver, hopefully the tests are enabled again once they run from a cloud provider repo
